고양이 장난감가게 유효성 추가하기, 회원 관리 만들기

kckoh opened the pull request:
안녕하세요! 
열심히 해보겠습니다. 
차근차근 기능 추가하면서 리뷰 받으면서 하겠습니다.

kckoh committed:
fix: remote @NoArgsConstructors and @AllArgsConstructor

롬복의 AllArgsConstructor/NoArgsConstructor 제거

kckoh committed:
add: UserTest and User

User 클라스 구현/테스트 작성

ssisksl77 commented:
한 주간 잘 부탁드립니다!

kckoh committed:
test: UserTest Email validation

유저클라스 이메일 테스트 코드 작성 및 에러 코드 작성

kckoh committed:
test: add username validaiton test to UserTest

유저 이름 유효성 테스트 추가

kckoh committed:
feat: add assertion to username field in User class

username assertion 추가

kckoh committed:
test: add password validtion test to UserTest

패스워드 유효셩 검사 테스트 작성:x

ssisksl77 commented on app/src/main/java/com/codesoom/assignment/domain/User.java:
```diff
@@ -0,0 +1,48 @@
+@Entity
+public class User {
+
+    @NotBlank(message = UserErrorMessage.EMAIL_NULL)
+    @NotNull(message = UserErrorMessage.EMAIL_BLANK)
```
`MessageSource` 를 이용해서 message를 코드에서 분리해보는 것이 재밌는 경험이실 거에요!

kckoh replied:
`MessageSource`를 추가 했습니다.

ssisksl77 commented on app/src/test/java/com/codesoom/assignment/domain/UserTest.java:
```diff
@@ -0,0 +1,119 @@
+
+    }
+
+    @DisplayName("Username validation")
+    @Nested
+    class Describe_username {
+
+        @DisplayName("if null is given")
+        @Nested
+        class Context_no_username {
+
+            @DisplayName("throws illegal argument exception")
```
팀원이 영어에 익숙하다면 상관없지만 그렇지 않다면, 한글로 적는 것이 좋다고 생각해요!

kckoh replied:
전부 한글로 바꿨습니다!

kckoh committed:
fix: 영어에서 한글로 변경

ssisksl77 commented on app/src/test/java/com/codesoom/assignment/controllers/UserControllerTest.java:
```diff
@@ -27,36 +30,43 @@ class UserControllerTest {
+                        .andDo(print());
             }
         }
+
+
+
+
+
+
```
불필요한 코드군요!

ssisksl77 commented:
ab5181cc7ba25fe9f593b5cd9864102d727f6276 의 커밋명과 실제 변경된 내용과는 거리가 있는 것 같습니다!

kckoh committed:
feat: Messagesource 추가

kckoh committed:
fix: UserValidationErrorAdvice 추가/ MessageSource bean 추가

kckoh committed:
test: usercontroller create 메소드 테스트 작성

kckoh committed:
fix: messagesource 에러 수정

messages.properties/UserData에 typo 수정

kckoh committed:
feat: usercontroller create method 구현

kckoh committed:
test: UserService Create method 테스트 코드 작성

kckoh committed:
feat: UserService 클라스 구현

UserServiceTest	-> userRepository 모킹
UserService -> create method 구현
UserData -> toEntity/fromEntity method 생성
UserRepository/JpaUserRepository -> save() 생성

ssisksl77 commented:
55de4ce5df5280d706c13db93f154a836a88f9ab 는 커밋명과 변경된 내용보다 범위가 큽니다!

ssisksl77 commented on app/src/test/java/com/codesoom/assignment/application/UserServiceTest.java:
```diff
@@ -0,0 +1,52 @@
+
+    @BeforeEach
+    void setUp() {
+        userService = new UserService(userRepository);
+        given(userRepository.save(any(User.class))).will(invocation -> {
+            User source = invocation.getArgument(0);
+            return source;
+        });
```
이 코드에 대해 한번 부연설명을 해주시면 감사하겠습니다!

ssisksl77 commented on app/src/test/java/com/codesoom/assignment/application/UserServiceTest.java:
```diff
@@ -0,0 +1,52 @@
+
+    private UserRepository userRepository = mock(UserRepository.class);
+
+//
+
```
불필요한 코드군요!

ssisksl77 commented on app/src/main/java/com/codesoom/assignment/controllers/UserController.java:
```diff
@@ -0,0 +1,31 @@
+@RequestMapping("/users")
+public class UserController {
+
+    private UserService userservice;
```
nit : `final` 을 붙이는 것이 좋겠어요!

ssisksl77 commented on app/src/main/java/com/codesoom/assignment/dto/UserData.java:
```diff
@@ -0,0 +1,56 @@
+    }
+
+
+    public UserData() {
+    }
```
`username`, `password`, `email` 이 not null 이어야 하는 것으로 보이는데 이 생성자는 어떤 경우에 사용되나요?

ssisksl77 commented on app/src/main/java/com/codesoom/assignment/dto/UserRequestDto.java:
```diff
@@ -0,0 +1,31 @@
+    @Email
+    private String email;
+
+    @NotBlank
+    private Long password;
```
`Long` 일 때 `@NotBlank` 는 어떻게 행동하나요?

ssisksl77 commented:
고생하셨습니다. 회고를 공유해주세요!
